src/onnx/parse_slice.cpp

```diff
@@ -77,7 +86,39 @@ struct parse_slice : op_parser<parse_slice>
             op.axes = axes;
         }
 
-        return info.add_instruction(op, args[0]);
+        if(reverse_direction)
+        {
+
+            migraphx::argument axes_arg = args.at(3)->eval();
+            std::vector<int> axes_v;
+            axes_arg.visit([&](auto s) { axes_v.assign(s.begin(), s.end()); });
+
+            auto lens = args[0]->get_shape().lens();
+
+            for(auto axis : axes_v)
+            {
+                auto start_v = op.starts[axis];
+                auto end_v   = op.ends[axis];
+                if((start_v < 0) && (end_v < INT_MIN))
+                {
+                    op.ends[axis]   = lens[axis] + start_v + 1;
+                    op.starts[axis] = 0;
+                }
```

Comment:
Why are you normalizing the start and ends? This seems unnecessary.

Reply:
one of the cases was start = -1 and end = INT_MIN, this caused issues (miscalculation the boundaries from slice op implementation) so I have normalized it, as if it is slice op with step=1

Reply:
I think we can just swap those values:

```cpp
for(auto axes:raxes)
    std::swap(op.starts[axes], op.ends[axes]);
```

It should use the same normalization that slice already uses, if not, then we should fix our normalization.

Reply:
The slice start and ends works as [starts, end), i.e., including starts, execluding ends. so, for positive steps, starts (smaller indices) are included, ends(larger values) are  excluded. For negative steps, starts(larger indices) are included and ends (smaller indices) are excluded, so we cannot just do a swap.  If all starts and ends are positive, we need to add one after swapping. 

For negative starts and ends, the scenarios is more complex. For example, -1 indicates the last dimension, i.e. dim_size -1. If we add 1 directly, it becomes 0, which is the first dimension.